### PR TITLE
B:18828 - Add Create Okta Account Logic to UpdateRequestedOfficeUser(MERGE AFTER B-18821 & B-18825)(READY)

### PR DESCRIPTION
**Pull request by TevinAdams**
## Summary

MERGE AFTER [B-18821](https://github.com/transcom/mymove/pull/12281) - MERGED
MERGE AFTER [B-18825](https://github.com/transcom/mymove/pull/12288) - MERGED

This PR introduces additions to the UpdateRequestedOfficeUserHandler. These additions includes the following:
1. Create okta Account Logic - (This only happens in the case that param.Status == "APPROVED")
2. UpdateRequestedOfficeUser will always run in all cases EXCEPT the case when CreateAccountOkta logic runs and fails.

[Okta API Specifications](https://developer.okta.com/docs/reference/api/roles/) explains more about the okta api specifications

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Start the client & server and log in as a MM Admin
2. Using postman or swagger UI
3. Send a PATCH request to /requested-office-users/{officeUserId} (Make sure Status is set to "APPROVED")
4. Provide relevant needed data for Okta account creation such as otherUniqueId, Email, Firstname, Lastname, gsaID, EDIPI, MobilePhone

Example: May need to adjust for your local env
{
  "firstName": "Leo",
  "middleInitials": "J",
  "lastName": "Spaceman",
  "email": "user5435@userdomain.com",
  "telephone": "212-555-5555",
  "roles": [
     {
      "name": "Transportation Invoicing Officer",
      "roleType": "transportation_invoicing_officer"
    }
  ],
  "transportationOfficeId": "f344dde8-1c04-4a37-b66d-3f1bab7c494c",
  "edipi": "1111111110",
  "otherUniqueId": "1111111110",
  "rejectionReason": "string",
  "status": "APPROVED"
}

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: swagger-def/admin.yaml [hunk 1]
```diff
@@ -1201,7 +1219,7 @@ paths:
         $ref: paths/office-users/get/description.md
       operationId: indexRequestedOfficeUsers
       tags:
-        - Requested office users
+        - Requested Office Users
```
Comment (cameroncaci) [inline swagger-def/admin.yaml:1222]: @TevinAdams 
The chances of the requested office user text wasn't present on the int branch. Was this added during a merge conflict resolution?

Comment (TevinAdams) [reply swagger-def/admin.yaml:1222]: @cameroncaci  I added this because of Spectral linter warnings causing the pipeline to fail.

Comment (danieljordan-caci) [reply swagger-def/admin.yaml:1222]: In `main` it is `Requested office users` but in this PR it is `Requested Office Users`

**Review by cameroncaci — APPROVED** [review]
Thanks for getting those int commits in here. G2g

**Review by danieljordan-caci — APPROVED** [review]

**Review by traskowskycaci — APPROVED** [review]
Looks good, only difference I saw was the Requested Office Users but checked int and Requested Office Users is what is in int so good to match them
https://github.com/transcom/mymove/blob/integrationTesting/swagger-def/admin.yaml

File: swagger-def/admin.yaml (hunk 1 above)
Comment (TevinAdams) [reply swagger-def/admin.yaml:1222]: @danieljordan-caci  Their tag is `Requested Office Users`. This was causing spectral linter warnings to go up which was causing my pipeline to fail so I made them match their tag.

Comment (danieljordan-caci) [reply swagger-def/admin.yaml:1222]: Yup I'm tracking! Thanks!

**Comment by traskowskycaci** [comment]
> Looks good, only difference I saw was the Requested Office Users but checked int and Requested Office Users is what is in int so good to match them https://github.com/transcom/mymove/blob/integrationTesting/swagger-def/admin.yaml

Actually the tags in int are still lowercase, I was just looking at the name

**Review by r-mettler — APPROVED** [review]
Changes same in INT with exception of Requested Office Users vs Requested office users, which is due to spectral linter warnings and has been resolved.

**Pull request merged**